include/dfu/suit_dfu_fetch_source.h

```diff
@@ -65,37 +72,38 @@ typedef int (*suit_dfu_fetch_source_request_fn)(const uint8_t *uri, size_t uri_l
 /**
  * @brief Register fetch source
  *
- * @param[in]   request_fn		Fetch source callback function. The suit processor will
- * iterate over registered callback functions when fetching from an URI is needed.
- * Attention! Order of execution of callback functions is not guaranteed!
- *
+ * @param[in] request_fn  Fetch source callback function. The suit processor will iterate over
+ *                        registered callback functions when fetching from an URI is needed.
+ *                        Attention! Order of execution of callback functions is not guaranteed!
  *
- * @return 0 on success, negative error code otherwise
+ * @return 0 on success, negative error code otherwise.
  */
 int suit_dfu_fetch_source_register(suit_dfu_fetch_source_request_fn request_fn);
 
 /**
  * @brief Passes the fetched data to the SUIT processor.
- *        This function shall only be called from inside a suit_dfu_fetch_source_request_fn
- *        callback.
  *
- * @param[in]   session_id Session id, the same which was passed to the session_id parameter of the
- *                         suit_dfu_fetch_source_request_fn callback.
- * @param[in]   data       Pointer to the data to write.
- * @param[in]   len        Length the data to write.
+ * @details
+ * This function shall only be called from inside a @ref suit_dfu_fetch_source_request_fn callback.
```

Comment:
Does this requirement force the callback to be a blocking function executing until the download process is completed? I think it should be clearly described in the callback description. So far I had an impression that callback can end early and continue downloading in a lower priority thread or workqueue.

Reply:
Correct, it is blocking, as pointed by the return code:
`0 on success - if fetching from the URI is possible and writing the data succeeded`

Reply:
The return code can be interpreted in many ways. "writing the data succeeded" can be interpreted as one chunk of a series. I would add to the callback function docs a clear expectation that the callback does not return until the download process is completed or failed.

Reply:
Added o note about the fact that the callback function shall block execution for the whole download process.